core/src/main/java/org/mule/runtime/core/streaming/PagingProviderProducer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.streaming;
+
+import static org.mule.runtime.core.config.i18n.MessageFactory.createStaticMessage;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.connection.ConnectionHandler;
+import org.mule.runtime.core.api.MuleRuntimeException;
+import org.mule.runtime.core.api.connector.ConnectionManager;
+import org.mule.runtime.api.streaming.PagingProvider;
+import org.mule.runtime.extension.api.runtime.ConfigurationInstance;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * Implementation of {@link Producer} that uses an instance of {@link PagingProvider} to get its results.
+ *
+ * @param <T> the type of the elements returned by the {@link PagingProvider}.
+ * @since 3.5.0
+ */
+public class PagingProviderProducer<T> implements Producer<List<T>> {
```

Comment:
this class should be in extensions-support, since it's a producer which works based on an interface defined in extensios-api